rclcpp/src/rclcpp/context.cpp

```diff
@@ -35,21 +35,91 @@
 
 #include "./logging_mutex.hpp"
 
-/// Mutex to protect initialized contexts.
-static std::mutex g_contexts_mutex;
-/// Weak list of context to be shutdown by the signal handler.
-static std::vector<std::weak_ptr<rclcpp::Context>> g_contexts;
+using rclcpp::Context;
+
+namespace rclcpp
+{
+/// Class to manage vector of weak pointers to all created contexts
+struct WeakContextsWrapper
```

Comment:
@ivanpauno nit: why not a `class`?

Reply:
Yes, you're right.

I originally was going to use a `struct`.
Then I thought that a `class` was more elegant, but forgot to update the keyword.

Fixed in https://github.com/ros2/rclcpp/pull/1132/commits/2b081fc7b26b74c4a25da7c1b9df8844fcd7702b.